Orange/widgets/evaluate/owpredictions.py

```diff
@@ -291,19 +293,37 @@ def _update_scores(self):
                 continue
             row = [QStandardItem(learner_name(pred.predictor)),
                    QStandardItem("N/A"), QStandardItem("N/A")]
+
+            actual = results.actual
+            predicted = results.predicted
+            probabilities = results.probabilities
+
+            mask = numpy.isnan(results.actual)
+            no_targets = mask.sum() == len(results.actual)
+            results.actual = results.actual[~mask]
+            results.predicted = results.predicted[:, ~mask]
+            results.probabilities = results.probabilities[:, ~mask]
```

Comment:
Although it seems that the only potential source of exceptions is handled, I think I'd feel better if this modifying and restoring of `results` was secured with `try-finally`. In the future, somebody might add more code in between that could raise unhandled exceptions. `try-finally` blocks (especially `finally`) would also visually show what it going on here.

Reply:
Lint doesn't aggree with that.

Reply:
I agree with lint. :) Remove `except`, keep just `finally`. We don't want to mask the exception, the idea was just ensure that `result` is restored.